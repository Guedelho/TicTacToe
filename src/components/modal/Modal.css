@import url("https://fonts.googleapis.com/css?family=Lato:300,400,700");
body {
  margin: 0;
  font-family: 'Lato', sans-serif; }

* {
  box-sizing: border-box; }
  *:focus {
    outline: none; }

.hide {
  display: none; }

.Modal {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 9;
  background: #fff; }
  .Modal:first-child {
    z-index: 10; }
  .Modal.player1 .content {
    border-color: #007bff; }
  .Modal.player1 input {
    border-color: #007bff;
    color: #007bff; }
  .Modal.player1 button {
    background: #007bff; }
  .Modal.player1 h3 {
    color: #007bff; }
  .Modal.player2 .content {
    border-color: #28a745; }
  .Modal.player2 input {
    border-color: #28a745;
    color: #28a745; }
  .Modal.player2 button {
    background: #28a745; }
  .Modal.player2 h3 {
    color: #28a745; }
  .Modal .content {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 90%;
    padding: 20px;
    max-width: 400px;
    text-align: center;
    border: thin solid #343a40;
    border-radius: 20px;
    transform: translate(-50%, -50%);
    background: #fff; }
  .Modal input {
    display: block;
    margin: 10px auto;
    padding: 5px 10px;
    font-size: 16px;
    border: thin solid #343a40;
    border-radius: 10px; }
  .Modal button {
    display: block;
    margin: 20px auto 0;
    font-size: 16px;
    background: #343a40;
    border: 0;
    color: #fff;
    padding: 5px 15px;
    border-radius: 15px; }
