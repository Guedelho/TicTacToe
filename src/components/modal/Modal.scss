@import '../../assets/style.scss';

.Modal {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: 9;
  background:$white;
  &:first-child{
    z-index: 10;
  }
  &.player1{
    .content {
      border-color: $blue;
    }
    input {
      border-color: $blue;
      color: $blue;
    }
    button {
      background: $blue;
    }
    h3 {
      color: $blue;
    }
  }
  &.player2{
    .content {
      border-color: $green;
    }
    input {
      border-color: $green;
      color: $green;      
    }
    button {
      background: $green;
    }
    h3 {
      color: $green;
    }
  }
  .content {
    position: absolute;
    top: 50%;
    left: 50%;
    width: 90%;
    padding: 20px;
    max-width: 400px;
    text-align: center;
    border: thin solid $black;
    border-radius: 20px;
    transform: translate(-50%, -50%);
    background: $white;
  }
  input {
    display: block;
    margin: 10px auto;
    padding: 5px 10px;
    font-size: 16px;
    border: thin solid $black;
    border-radius: 10px;
  }
  button {
    display: block;
    margin: 20px auto 0;
    font-size: 16px;
    background: $black;
    border: 0;
    color: $white;
    padding: 5px 15px;
    border-radius: 15px;
  }
}