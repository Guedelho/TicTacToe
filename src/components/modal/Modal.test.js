import React from 'react';
import { shallow } from 'enzyme';
import Modal from './Modal';

const props = {
  title: "string",
  name: "string",
  playerName: "string",
  onChangePlayerName: jest.fn(),
}

it('renders without crashing', () => {
  expect(shallow(<Modal {...props}/>)).toMatchSnapshot();
});