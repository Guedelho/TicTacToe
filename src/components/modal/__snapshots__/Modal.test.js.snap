// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Modal
    name="string"
    onChangePlayerName={[Function]}
    playerName="string"
    title="string"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <form
        className="content"
        onSubmit={[Function]}
>
        <h3>
                string
        </h3>
        <input
                maxLength="8"
                minLength="2"
                name="string"
                onChange={[Function]}
                required={true}
                value="string"
        />
        <button
                type="submit"
        >
                Continuar
        </button>
</form>,
      "className": "Modal string",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            string
</h3>,
          <input
            maxLength="8"
            minLength="2"
            name="string"
            onChange={[Function]}
            required={true}
            value="string"
/>,
          <button
            type="submit"
>
            Continuar
</button>,
        ],
        "className": "content",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "string",
          },
          "ref": null,
          "rendered": "string",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "maxLength": "8",
            "minLength": "2",
            "name": "string",
            "onChange": [Function],
            "required": true,
            "value": "string",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Continuar",
            "type": "submit",
          },
          "ref": null,
          "rendered": "Continuar",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <form
          className="content"
          onSubmit={[Function]}
>
          <h3>
                    string
          </h3>
          <input
                    maxLength="8"
                    minLength="2"
                    name="string"
                    onChange={[Function]}
                    required={true}
                    value="string"
          />
          <button
                    type="submit"
          >
                    Continuar
          </button>
</form>,
        "className": "Modal string",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h3>
              string
</h3>,
            <input
              maxLength="8"
              minLength="2"
              name="string"
              onChange={[Function]}
              required={true}
              value="string"
/>,
            <button
              type="submit"
>
              Continuar
</button>,
          ],
          "className": "content",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "string",
            },
            "ref": null,
            "rendered": "string",
            "type": "h3",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "maxLength": "8",
              "minLength": "2",
              "name": "string",
              "onChange": [Function],
              "required": true,
              "value": "string",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Continuar",
              "type": "submit",
            },
            "ref": null,
            "rendered": "Continuar",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
